src/Clients/BaseClient.php

```diff
@@ -26,4 +48,43 @@ public function createFilterQuery(string $path, array $filters): string
 
         return "{$path}?{$queryString}";
     }
+
+    public function getCacheTtl(): int
+    {
+        return config('easyoffice-api.default_cache_liftime');
+    }
+
+    public function getCacheEnabled(): bool
+    {
+        return config('easyoffice-api.enable_cache', false);
+    }
+
+    public function generateCacheKey(string $endpoint, array $options = []): string
+    {
+        $cacheKey = "easyoffice__{$endpoint}";
+
+        foreach ($options as $key => $value) {
+            $cacheKey .= "__{$key}_{$value}";
+        }
+
+        return $cacheKey;
+    }
+
+    public function handleErrors(Response $response): void
+    {
+        if ($response->status() === 422) {
+            $messages = $response->json('errors') ?? [];
+
+            throw ValidationException::withMessages($messages);
+        }
+
+        Log::error("[EO] API call to EasyOffice failed with status code {$response->status()}", [
+            'reason' => $response->reason(),
+            'response' => $response->body(),
+            'headers' => $response->headers(),
+            'requestUri' => $response->transferStats?->getRequest()?->getUri(),
+        ]);
+
+        abort($response->status(), $response->reason());
```

Comment:
I'm not sure if the package is the correct place to do this. Wouldn't we want the flexibility to handle this differently foreach project?

Proposal: move this logic to the project level, where you could overwrite the container binding and add this? What do you think?